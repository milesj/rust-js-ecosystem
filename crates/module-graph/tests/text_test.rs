mod utils;

use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::generate_graph_for_file;

mod text {
    use super::*;

    #[test]
    fn graphql() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "graphql.gql"));
    }

    #[test]
    fn graphql_alt_extension() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "graphql.graphql"));
    }

    #[test]
    fn html() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "index.html"));
    }

    #[test]
    fn less() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.less"));
    }

    #[test]
    fn map() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "source.js.map"));
    }

    #[test]
    fn sass() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.sass"));
    }

    #[test]
    fn scss() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.scss"));
    }

    #[test]
    fn stylus() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.styl"));
    }

    #[test]
    fn svg() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "icon.svg"));
    }
}
