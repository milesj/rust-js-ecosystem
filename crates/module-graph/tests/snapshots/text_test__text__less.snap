---
source: crates/module-graph/tests/text_test.rs
expression: "generate_graph_for_file(sandbox.path(), \"styles.less\")"
---
{
    1: Module {
        exports: [],
        fragment: None,
        id: 1,
        imports: [],
        package_name: None,
        path: "/root/styles.less",
        query: None,
        source: TextModule {
            kind: Less,
            source: "@width: 10px;\n@height: @width + 10px;\n\n#header {\n  width: @width;\n  height: @height;\n}\n\n.bordered {\n  border-top: dotted 1px black;\n  border-bottom: solid 2px black;\n}\n\n#menu a {\n  color: #111;\n  .bordered();\n}\n\n.post a {\n  color: red;\n  .bordered();\n}\n",
        },
    },
}
